### DISCUSSION TITLE
GitHub CoPilot for Visual Studio 2022 17.6.0 Preview 2.0 hangs Visual Studio

### DISCUSSION THREAD

**Discussion by digitaldias** [community/community#50411] [Copilot Conversations]
### Select Topic Area

Bug

### Body

After updating my Visual Studio 2022 preview instance to the latest, I found that opening any document leads to Visual studio hanging, and only being able to be shut down via Task Manager (or similar). 

It will load up the solution, and display the solution explorer, but freeze on opening any document. 

I suspect the CoPilot authorization dialog never gets displayed and hangs as a modal somewhere out of my reach. 
The CoPilot version that I am using is 1.77.1.0

We are on a paid company subscription to CoPilot and would love to see this go away soon, even though we are on a preview release of Visual Studio 2022.

**Comment by ayim** [comment]
Hey, so sorry you're running into problems with getting started.

We're super interested in doing a screenshare/live debugging this because getting good repros can be hard in the AI world. Please please shoot me an email at [username]@github.com and we'll get the ball rolling.

**Comment by Kingside88** [comment]
I have the same issue. Starting a solution under VS 17.5.3 and Github Copilot freezes the studio.

**Reply by ayim** [reply]
Hey @Kingside88, we don't like the sound of freezing the studio either.

We're super interested in doing a screenshare/live debugging this because getting good repros can be hard in the AI world. Please please shoot me an email at [username]@github.com and we'll get the ball rolling.

**Comment by FranciscoBucioDTI** [comment]
I have the same issue. I'm using Visual Studio 2022 Enterprise version 17.5.3 and Copilot 1.77.2.0 and VS2022 becomes unresponsive when opening a file.

However, I've been able to find a workaround, maybe it helps you finding the error.

When I open VS, nothing shows under "Output window", as if GitHub Copilot service hasn't started yet:
![image](https://user-images.githubusercontent.com/110624108/226869590-4ff1909a-ef93-4989-9d99-60f410699858.png)

The moment I select "Github" at Tools - Options, "Output window" displays info about GitHub Copilot service starting:
![image](https://user-images.githubusercontent.com/110624108/226869645-1ce5a776-f217-41a9-ab6a-af5c86d37894.png)

After that, I have no problem opening any file, it doesn't freezes anymore.

**Reply by Transigent** [reply]
Yes I replicated the exact same behaviour, thanks. I did exactly these steps and suddenly the activation process that requires entry of the authorization code on Github's activation page worked. Thanks a lot.

EDIT:   It looks like I have to do this every time I start Visual Studio. And, if from the Start dialog I choose a project that already had a file(s) open for editing before I last shut it down, it will hang when the solution opens and it tries to load the file. So instead I have to Continue Without Code on the start dialog and then do the above procedure, and only then open the solution.

**Reply by ayim** [reply]
@Transigent @FranciscoBucioSTI highly interesting, I haven't seen this failure mode before.

Please contact me at ayim at github dot com so that we can screenshare/collect logs together and investigate - this will be super helpful to get a fix for this rolling.

**Comment by dcarapic** [comment]
Any fix for this? It also happens for me as well:
GitHub Copilot 1.77.2.0 (v1.77.2.0@570999d7d)
Visual Studio Professional 2022 (64-bit) Version 17.5.3

**Reply by ayim** [reply]
Hey @dcarapic, Aaron from Copilot team here... we need your help to develop a fix for this.

We're super interested in doing a screenshare/live debugging this because getting good repros can be hard in the AI world. Please shoot me an email at [username]@github.com and we'll get the ball rolling.

**Comment by Transigent** [comment]
I'm having exactly the same issue. I just finished installing Windows a couple of days ago, I thought I would try VS2022 with Copilot. Straight after installing I installed Copilot extension.

Then I tried to open a project, which worked. It showed files in the Solution Explorer sidebar but when I tried to run, it compiled then hung, after a few seconds presumably when it hit a breakpoint and tried to display the code file. 

I restarted and tried to open a code file. It hung again.

I uninstalled VS2022 and reinstalled from scratch. This time I did not install Copilot. Everything worked, I compiled and ran no problems.

I then installed Copilot and tried to create a new project. It seemed to work but hung again at this screen:

<img width="775" alt="image" src="https://user-images.githubusercontent.com/18422245/227178613-e458a57a-79e7-4ebf-bd99-49a19d53e68e.png">

It wasnt a proper 'hang' because the green segment was still scrolling, you could also drag around the "Creating project..." dialog but no buttons worked and I had to close it from Task Manager.

If I disable Copilot by clicking Continue Without Code on the start dialog, then Manage Extensions and disable Copilot, I can do everything normally again, create projects, open existing ones. 

It really does seem like it's related to the opening of a code file in the solution. Everhthing in the VS interface seems OK until you try to open a file.

VS 2022 Version 17.5.3
Copilot 1.77.2.0

**Reply by phydelta** [reply]
I have exactly same situation.

VS 2022 Version 17.5.3
Copilot 1.77.2.0

**Reply by ayim** [reply]
@phydelta Aaron from Copilot team here... we need your help to develop a fix for this.

We're super interested in doing a screenshare/live debugging this because getting good repros can be hard in the AI world. Please shoot me an email at [username]@github.com and we'll get the ball rolling.

**Comment by clusterzx** [comment]
Same phenomen with my installation. The workaround from @FranciscoBucioSTI did work for me.
Latest VS 2022 Community + latest CoPilot Extension Version.

**Comment by radek7210** [comment]
Same problem here, reproducible in 100 % cases:
- use VS, edit a file
- leave the file open in the editor, close VS
- in Settings, I have the option ticked to re-open previous documents on startup

Then, on next VS startup, the VS hangs.
Workaround: close all documents before exiting VS (or change the Settings to not re-open documents on startup)

**Reply by ayim** [reply]
Hey @radek7210, Aaron from Copilot team here - tried to repro on my machine but didn't run into the issue.... so we'd like your help to develop a fix for this.

We're super interested in doing a screenshare/live debugging this because getting good repros can be hard in the AI world. Please shoot me an email at [username]@github.com and we'll get the ball rolling.

**Comment by MiddleKerb** [comment]
Same problem here, workaround from @FranciscoBucioSTI helped but i need to do that every time I start Visual Studio
VS 2022 Version 17.5.3
Copilot 1.77.2.0

**Comment by jdharte** [comment]
Wow, same problem here.  The productivity gains of Copilot are being negated by this.  Where's the fix?

VS 2022 Version 17.4.6
Copilot 1.77.2.0

**Comment by ipc-foundry1** [comment]
I am having the exact same problem. A fix would be greatly appreciated - this is really frustrating. 
VS 2022 Version 17.5.3
Copilot 1.77.2.0

**Comment by FranciscoBucioDTI** [comment]
I don't know about you guys, but with me the problem seems to be fixed since Friday or Saturday (don't remember exactly).

Now I can open any project/file directly, no need for the previous workaround anymore.

**Reply by radek7210** [reply]
I saw the hang only once yesterday and once today. Don't remember if it was the production version of VS or the VS preview. However - clearly, something has significantly improved - the hangs used to happen in 100 % case, while now they are very rare and in majority cases the solution opens just fine. The version of the extension seems to be the same though (1.77.2.0). Has something changed on the server side?

**Reply by jdharte** [reply]
I thought for a second it might have been fixed for me, so I turned on the reopen documents option, restarted studio, and it hung opening my solution.  Then I turned off the reopen documents option, restarted studio, opened the solution and tried to open a .tsx file and it locked up.  Definitely still broken.  Feels like a race condition somewhere...

**Comment by radek7210** [comment]
not sure if this helps - this is what VS shows in a Dump taken during the hang:

![image](https://user-images.githubusercontent.com/41726008/228138799-eae6ff91-aba4-4554-b0a8-55ad72330424.png)

**Reply by radek7210** [reply]
![image](https://user-images.githubusercontent.com/41726008/228141459-af5732ae-f471-43b0-97ff-72eadaaeb552.png)

**Comment by RichardSinden** [comment]
This only seems to be affecting my WPF projects. If VS tries to open a .xaml file before it opens a .cs file then VS will hang (either on startup when loading from a previous session, or manually after loading solution with no documents open). I can reproduce this with a new WPF project.
VS 2022 Version 17.5.3
Copilot 1.77.2.0

**Comment by bondarenkod** [comment]
Following the installation of the Co-Pilot extension, VS 22 Enterprise functioned flawlessly for a brief period. However, subsequent installation of the ResX Manager extension caused the software to experience hang-ups. Despite attempts to remedy the situation by removing the extension, the issue persisted. Resuming functionality was only possible upon disabling the Co-Pilot extension.

**Comment by radek7210** [comment]
Version v1.79.0 installed, the issues are gone. Thx.
